imports/plugins/core/layout/client/components/coreLayout.js

```diff
@@ -16,17 +16,28 @@ const CoreLayout = ({ actionViewIsOpen, structure }) => {
   const headerComponent = layoutHeader && getComponent(layoutHeader);
   const footerComponent = layoutFooter && getComponent(layoutFooter);
 
+  let mainNode = null;
+  if (Template[template]) {
+    mainNode = <Blaze template={template} />;
+  } else {
+    try {
+      const mainComponent = getComponent(template);
+      mainNode = React.createElement(mainComponent, {});
+    } catch (error) {
+      // no-op
```

Comment:
We should be checking for the React component first, and in the `catch` statement try getting the blaze component instead. This way we don't have a blank `catch` statement, and we assume that most components in the app are React.

Reply:
@mikemurray The ordering was intentionally to ensure that the current behavior is retained and minimizing risk of breaking changes. That's the reason the Blaze template was probed first. But I've also been thinking that we should make the React component the first class citizen, not vice versa. But this would introduce a slightly higher risk to break user plugins (E.g. if a user has registered a React component with the same name like the Blaze template name).